src/DynamicData/List/Internal/BufferIf.cs

```diff
@@ -72,7 +72,7 @@ public IObservable<IChangeSet<T>> Run()
                                                        }
 
                                                        observer.OnNext(new ChangeSet<T>(buffer));
-                                                       buffer.Clear();
+                                                       buffer = new List<Change<T>>();
```

Comment:
I wonder if `ChangeSet<T>`'s constructor should be generating a new list in the `List<T>` based parameter. At the moment it just sets a reference, and does ToList() only in the non-`List<T>` parameter

Reply:
I felt like there might be other scenarios where you know the list you're giving the ChangeSet is safe to use directly, so we would not want it to make another copy again right after for no reason.

The way it is now leaves it more up to the creator of the ChangeSet<T> to decide if/when the copy needs to happen, which I think is more appropriate.

Reply:
@glennawatson `ChangeSet<T>` needs to be optimised - the list copying is not desirable. A while back, I changed `ChangeSet<TObject, TKey>` so it is inherited directly from a list.  The same needs to be applied across the board for list functions

Reply:
`buffer = new List<Change<T>>();` makes sense